pkg/clusters/addons/argocd/addon.go

```diff
@@ -0,0 +1,176 @@
+package argocd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/kong/kubernetes-testing-framework/internal/retry"
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+)
+
+const (
+	// AddonName indicates the unique name of this addon.
+	AddonName clusters.AddonName = "argocd"
+
+	// DefaultNamespace indicates the default namespace this addon will be deployed to.
+	DefaultNamespace = "argocd"
+
+	canaryDeployment = "argocd-repo-server"
+
+	manifestURL = "https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/core-install.yaml"
```

Comment:
As `stable` sounds like a reasonable default, I suggest we also allow using a pinned version avoid unexpected changes kicking in with no further ado. I'd expect us to pin this version in any of our tests.
```suggestion
	manifestURLTemplate = "https://raw.githubusercontent.com/argoproj/argo-cd/%s/manifests/core-install.yaml"
	defaultTag = "stable"
```
The builder could have `WithVersion` method to allow using versions different than `stable`.

Reply:
👍 for using a pinned version. We already had issues with a floating version pointer (like a branch) in the past.

Reply:
Borrowing the configurable approach from Istio, minus the lookup (since Argo provides their own "latest stable". You get a floating tag by default, but can `WithVersion()` to pin one as a KTF client, as a balance between stability and not needing to do a new release to get updates.